easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.13.0-foss-2023a-CUDA-12.1.1.eb

```diff
@@ -0,0 +1,250 @@
+easyblock = 'PythonBundle'
+
+name = 'TensorFlow'
+version = '2.13.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://www.tensorflow.org/'
+description = "An open-source software library for Machine Intelligence"
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'pic': True}
+
+builddependencies = [
+    ('Bazel', '6.3.1'),
+    # git 2.x required, see also https://github.com/tensorflow/tensorflow/issues/29053
+    ('git', '2.41.0', '-nodocs'),
+    ('pybind11', '2.11.1'),
+    ('UnZip', '6.0'),
+    # Required to build some of the extensions
+    ('poetry', '1.5.1'),
+    # System protobuf doesn't seem to work: https://github.com/tensorflow/tensorflow/issues/61593
+    # So don't add it here
+]
+dependencies = [
+    ('CUDA', '12.1.1', '', SYSTEM),
+    ('cuDNN', '8.9.2.26', versionsuffix, SYSTEM),
```

Comment:
@VRehnberg It seems like this `cuDNN` version is causing trouble, I'm getting:

```
In file included from bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_Operation.h:37,
                 from bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_OperationGraph.h:36,
                 from bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_Heuristics.h:31,
                 from bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend.h:101,
                 from tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:56:
bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_PointWiseDesc.h: In member function int64_t cudnn_frontend::PointWiseDesc_v8::getPortCount() const:
bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_PointWiseDesc.h:69:16: error: enumeration value CUDNN_POINTWISE_RECIPROCAL not handled in switch [-Werror=switch]
   69 |         switch (mode) {
      |                ^
bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_Operation.h: In member function cudnn_frontend::Operation_v8&& cudnn_frontend::OperationBuilder_v8::build_pointwise_op():
bazel-out/k8-opt/bin/external/cudnn_frontend_archive/_virtual_includes/cudnn_frontend/third_party/cudnn_frontend/include/cudnn_frontend_Operation.h:413:16: error: enumeration value CUDNN_POINTWISE_RECIPROCAL not handled in switch [-Werror=switch]
  413 |         switch (m_operation.pointwise_mode) {
      |                ^
cc1plus: some warnings being treated as errors
```

see also https://github.com/tensorflow/tensorflow/issues/60832, where they suggest to downgrade to an older `cuDNN` (ugh...)

Reply:
There are no easyconfigs yet that using a `2023a` toolchain and have a `cuDNN` dependency, so we still have the freedom to stick to `cuDNN` 8.6.* here...

Reply:
We also need to stick to CUDA 11.8 though, since cuDNN 8.6 is only paired with CUDA 10.3 and 11.8 it seems, see https://developer.download.nvidia.com/compute/redist/cudnn/v8.6.0/local_installers/

Reply:
And CUDA 11.8 is a problem with GCC 12.x, hitting this when installing NCCL on top of CUDA 11.8.0 with `GCCcore/12.3.0`:
```
unsupported GNU version! gcc versions later than 11 are not supported!
```

So that tells me we're doomed to stick to `foss/2022a` for TensorFlow 2.13.0?

Reply:
Meh, I'll close this one then.

Reply:
Hmm, or go with another CUDA I suppose. That's what the CUDA version suffix is for I guess. For GCC 12.3 I can't find anything about compatible GCC, but extrapolating what I could find it will probably work for CUDA 12.3 which isn't [listed](https://docs.nvidia.com/deeplearning/cudnn/support-matrix/index.html) for CuDNN 8.9.6, but could possibly work.

Reply:
> ```
> unsupported GNU version! gcc versions later than 11 are not supported!
> ```
> 
> So that tells me we're doomed to stick to `foss/2022a` for TensorFlow 2.13.0?

We can workaround this somehow by forcing NVCC to accept the "incompatible" compiler: https://github.com/easybuilders/easybuild-easyconfigs/pull/18853/files#diff-c0833191974a98d7eddf20cecac9d27ec670e369f43f75f3a4bafb2261b1135fR27